pkg/restore/restore.go

```diff
@@ -474,11 +485,41 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 			if !existingNamespaces.Has(mappedNsName) {
 				logger := ctx.log.WithField("namespace", nsName)
 				ns := getNamespace(logger, filepath.Join(dir, api.ResourcesDir, "namespaces", api.ClusterScopedDir, nsName+".json"), mappedNsName)
-				if _, err := kube.EnsureNamespaceExists(ns, ctx.namespaceClient); err != nil {
+				nsIsReady, err := kube.EnsureNamespaceExistsAndIsReady(ns, ctx.namespaceClient)
+				if err != nil {
 					addArkError(&errs, err)
 					continue
 				}
 
+				if !nsIsReady {
+					timeout := time.After(ctx.resourceTerminatingTimeout)
+					tick := time.Tick(1 * time.Second)
+					var proceed bool
+					for proceed != true {
+						select {
+						case <-timeout:
+							err := errors.Errorf("Timed out waiting on NS %s to be ready", ns.Name)
+							addArkError(&errs, err)
+							return warnings, errs
+						case <-tick:
+							nsIsReady, err = kube.EnsureNamespaceExistsAndIsReady(ns, ctx.namespaceClient)
```

Comment:
I don't want to re-call `kube.EnsureNamespaceExistsAndIsReady` here. Let's implement my suggestion above unless you have objections?